server/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -122,9 +122,11 @@ public ExecutionPlan build(DependencyCarrier executor,
         LogicalPlan leftLogicalPlan = lhs;
         LogicalPlan rightLogicalPlan = rhs;
 
+        long lhsNumExpectedRows = lhs.numExpectedRows() < 0 ? 0 : lhs.numExpectedRows();
+        long rhsNumExpectedRows = rhs.numExpectedRows() < 0 ? 0 : rhs.numExpectedRows();
```

Comment:
```suggestion
        long rhsNumExpectedRows = Math.max(rhs.numExpectedRows(), 0);
```

But see previous comment, it may be better to skip changing the sides if any side has -1